core/src/main/java/com/linecorp/armeria/server/auth/AuthTokenExtractors.java

```diff
@@ -27,22 +27,52 @@
 public final class AuthTokenExtractors {
 
     /**
-     * A {@link BasicToken} extractor function instance.
+     * A {@link BasicToken} extractor function.
+     *
+     * @deprecated Use {@link #basic()}.
      */
+    @Deprecated
     public static final Function<RequestHeaders, BasicToken> BASIC =
             new BasicTokenExtractor(HttpHeaderNames.AUTHORIZATION);
 
     /**
-     * An {@link OAuth1aToken} extractor function instance.
+     * An {@link OAuth1aToken} extractor function.
+     *
+     * @deprecated Use {@link #oAuth1a()}.
      */
+    @Deprecated
     public static final Function<RequestHeaders, OAuth1aToken> OAUTH1A =
             new OAuth1aTokenExtractor(HttpHeaderNames.AUTHORIZATION);
 
     /**
-     * An {@link OAuth2Token} extractor function instance.
+     * An {@link OAuth2Token} extractor function.
+     *
+     * @deprecated Use {@link #oAuth2()}.
      */
+    @Deprecated
     public static final Function<RequestHeaders, OAuth2Token> OAUTH2 =
             new OAuth2TokenExtractor(HttpHeaderNames.AUTHORIZATION);
 
+    /**
+     * Returns a {@link BasicToken} extractor function.
+     */
+    public static Function<RequestHeaders, BasicToken> basic() {
```

Comment:
How about `Function<? super RequestHeaders, ? extends BasicToken>`

Reply:
Let me add `? super`. `? extends` is not necessary because it's final.